src/components/button/Button.js

```diff
@@ -31,7 +31,7 @@ Button.propTypes = {
   children: PropTypes.node.isRequired,
   disabled: PropTypes.bool,
   fullWidth: PropTypes.bool,
-  onClick: PropTypes.func.isRequired,
+  onClick: PropTypes.func,
```

Comment:
Do we need to make this optional? I feel like having a button that doesn't do anything is not a good UX xD

Reply:
I don't think we should enforce onClick prop on button component. 
There might be a case where I just want to pass the button as a trigger node to the panel without any onClick handler defined. And later on the Panel component adds its onClick functionality to the Button component.

Reply:
that sounds legit 👌

Reply:
I can't imagine the scenario by now. I would say to make it a mandatory parameter. If one day happens what you say, we will talk about this :thinking:

Reply:
@tureey 
Let's take a simple example - 
```
<Panel
    trigger={<Button text="Open Modal" />}
     closable={true}
     modal={true}
     open={this.state.open}>something</Panel>
```

Over here the `trigger` prop is a button. This button will have a onClick prop in the panel component, but not here. If Button has onClick as mandatory, then we will have to unnecessarily change it to `<Button onClick={noop} text='Open' />` which is a hack to overcome the limitation of Button component.

Reply:
My personal preference is to make it required as it was before.

Reply:
I do agree with you @crysfel